plugins/modules/digital_ocean_droplet.py

```diff
@@ -271,10 +285,27 @@ def get_droplet(self):
             json_data = self.get_by_name(self.module.params['name'])
         return json_data
 
+    def resize_droplet(self):
+        """API reference: https://developers.digitalocean.com/documentation/v2/#resize-a-droplet (Must be powered off)"""
+        if self.status == 'off':
+            response = self.rest.post('droplets/{0}/actions'.format(self.id),
+                                      data={'type': 'resize', 'disk': self.module.params['resize_disk'], 'size': self.module.params['size']})
+            json_data = response.json
+            if response.status_code == 201:
+                self.module.exit_json(changed=True, msg='Resized Droplet {0} ({1}) from {2} to {3}'.format(
+                    self.name, self.id, self.size, self.module.params['size']))
+            else:
+                self.module.fail_json(msg="Resizing Droplet {0} ({1}) failed [HTTP {2}: {3}]".format(
+                    self.name, self.id, response.status_code, response.json['message']))
+        else:
+            self.module.fail_json(msg='Droplet must be off prior to resizing (https://developers.digitalocean.com/documentation/v2/#resize-a-droplet)')
+
     def create(self):
         json_data = self.get_droplet()
         droplet_data = None
         if json_data:
+            if json_data['droplet']['size']['slug'] != self.module.params['size']:
```

Comment:
I don't like the indexing here either, but it's kind of annoying that the `slug` is three levels deep. I'd consider using `.get()` here too.

Reply:
So, turns out -- thanks for looking so closely and calling this out -- it's `size_slug` not `size.slug`!